File: BOJ_7568.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    pair<int, int> arr[N];
    vector<int> answer;

    for(int i = 0; i < N; i++) {
        cin >> arr[i].first >> arr[i].second;
    }

    for(int i = 0; i < N; i++) {
        int cnt = 0;
        for(int j = 0; j < N; j++) {
            if(arr[i] == arr[j]) continue;
            if(arr[i].second < arr[j].second && arr[i].first < arr[j].first) {
                cnt++;
            }
        }
        answer.push_back(cnt+1);
    }

    for(auto &i : answer) {
        cout << i << " ";
    }
}

File: BOJ_7568.kt
fun main() = with(System.`in`.bufferedReader()) {
    val N = readLine().toInt()
    val answer = Array(N) { 1 }
    val list = Array(N) {
        val (x, y) = readLine().split(" ").map { it.toInt() }
        Pair(x, y)
    }

    for(i in 0 until N) {
        for(j in 0 until N) {
            if(i == j) continue
            if(list[i].first < list[j].first && list[i].second < list[j].second) answer[i]++
        }
    }

    answer.forEach { print("$it ") }
}
